src/datatypes/map.c

```diff
@@ -27,9 +27,7 @@ static inline Pair Pair_New
 	SIValue val
 ) {
 	ASSERT(SI_TYPE(key) & T_STRING);
-	return (Pair) {
-		.key = SI_CloneValue(key), .val = SI_CloneValue(val)
-	};
+	return (Pair) { .key = key, .val = val };
```

Comment:
ensure this is safe

Reply:
Yeah, `Map` now exposes two functions for adding a key/value pair.
1. `Map_Add` clones (similar) to what we had.
2. `Map_AddNoClone` **new**, assumes ownership over the value, it's the caller's responsibility not to free the given values. at the moment the only call to `Map_AddNoClone` is done by `_SIValue_FromMapNode` which doesn't free the passed values.